lib/smb2/file.rb

```diff
@@ -4,29 +4,43 @@
 # An open file on the remote filesystem.
 class Smb2::File
 
-  # The tree that {Tree#create created} this File
-  #
-  # @return [Smb2::Tree]
-  attr_accessor :tree
+  # Some day we'll document all of these in one place
+  STATUS_END_OF_FILE = 0xC0000011
 
   # The server's response from when we {Tree#create created} this File
   #
   # @return [Smb2::Packet::CreateResponse]
   attr_accessor :create_response
 
+  # @return [String]
+  attr_accessor :filename
+
   # The last response we got from {#read}. Useful for figuring out what went
   # wrong.
   #
   # @return [Smb2::Packet::ReadResponse]
   attr_accessor :last_read_response
 
+  # Current offset of the read/write pointer from the beginning of the file,
+  # in bytes.
+  #
+  # @return [Fixnum]
+  attr_accessor :pos
+
+  # The tree that {Tree#create created} this File
+  #
+  # @return [Smb2::Tree]
+  attr_accessor :tree
+
   # @param tree [Smb2::Tree] the Tree where this File was opened. See {#tree}
   # @param create_response [Smb2::Packet::CreateResponse] the server's
   #   response from when we {Tree#create created} this File. See
   #   {#create_response}
-  def initialize(tree:, create_response:)
+  def initialize(filename:, tree:, create_response:)
+    self.filename = filename
     self.tree = tree
     self.create_response = create_response
+    self.pos = 0
```

Comment:
Thanks for using the attr_writers inside initialize instead of setting instance variables directly.  I prefer this style as it makes patching the code easier.